be/src/exec/vectorized/es_http_components.cpp

```diff
@@ -0,0 +1,540 @@
+        ss << "Expected value of type: " << type_to_string(type)          \
+           << "; but found type: " << json_type_to_raw_str(col.GetType()) \
+           << "; Docuemnt slice is : " << json_value_to_string(col);      \
+        return Status::RuntimeError(ss.str());                            \
+    } while (false)
+
+template <typename T>
+static Status get_int_value(const rapidjson::Value& col, PrimitiveType type, void* slot, bool pure_doc_value) {
+    return Status::OK();
+}
+
+ScrollParser::ScrollParser(bool doc_value_mode)
+        : _tuple_desc(nullptr),
+          _docvalue_context(nullptr),
+          _size(0),
+          _cur_line(0),
+          _doc_value_mode(doc_value_mode),
+          _temp_writer(_scratch_buffer) {}
+
+Status ScrollParser::parse(const std::string& scroll_result, bool exactly_once) {
+    _size = 0;
+    _cur_line = 0;
+    _document_node.Parse(scroll_result.data(), scroll_result.size());
+    if (_document_node.HasParseError()) {
+        return Status::InternalError(fmt::format("Parsing json error, json is: {}", scroll_result));
+    }
+
+    if (!exactly_once && !_document_node.HasMember(FIELD_SCROLL_ID)) {
+        LOG(WARNING) << "Document has not a scroll id field scroll reponse:" << scroll_result;
+        return Status::InternalError("Document has not a scroll id field");
+    }
+
+    if (!exactly_once) {
+        const rapidjson::Value& scroll_node = _document_node[FIELD_SCROLL_ID];
+        _scroll_id = scroll_node.GetString();
+    }
+
+    // { hits: { total : 2, "hits" : [ {}, {}, {} ]}}
+    const rapidjson::Value& outer_hits_node = _document_node[FIELD_HITS];
+    // if has no inner hits, there has no data in this index
+    if (!outer_hits_node.HasMember(FIELD_INNER_HITS)) {
+        return Status::OK();
+    }
+    const rapidjson::Value& inner_hits_node = outer_hits_node[FIELD_INNER_HITS];
+    // this happened just the end of scrolling
+    if (!inner_hits_node.IsArray()) {
+        return Status::OK();
+    }
+    _inner_hits_node.CopyFrom(inner_hits_node, _document_node.GetAllocator());
+    // how many documents contains in this batch
+    _size = _inner_hits_node.Size();
+
+    return Status::OK();
+}
+
+Status ScrollParser::fill_chunk(ChunkPtr* chunk, bool* line_eos) {
+    if (current_eos()) {
+        *line_eos = true;
+        return Status::OK();
+    }
+
+    *chunk = std::make_shared<Chunk>();
+    std::vector<SlotDescriptor*> slot_descs = _tuple_desc->slots();
+
+    // init column information
+    for (auto& slot_desc : slot_descs) {
+        ColumnPtr column = ColumnHelper::create_column(slot_desc->type(), slot_desc->is_nullable());
+        (*chunk)->append_column(std::move(column), slot_desc->id());
+    }
+
+    size_t left_sz = _size - _cur_line;
+    size_t fill_sz = std::min(left_sz, (size_t)config::vector_chunk_size);
+
+    auto slots = _tuple_desc->slots();
+
+    // TODO: we could fill chunk by column rather than row
+    for (size_t i = 0; i < fill_sz; ++i) {
```

Comment:
Why not fill chunk by column?

Reply:
Because there are too many checksum logics and branches inside this loop, filling the chunk by column is too complicated and requires extra space to store the values after json parsing.